AV

ShevchenkoAlexen opened the pull request:
перекоммитил

ShevchenkoAlexen committed:
перекоммитил

перекоммитил

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,244 @@
+/* Добавляем шрифт для заголовка*/
+@font-face {
+    font-family: 'COCOMAT-UltraLight';
+    src: url('COCOMAT-UltraLight.ttf.woff') format('woff'),
```
А откуда он берется, его же нет в проекте?

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,244 @@
+/* Если реальный чекбокс у нас отмечен, то тогда добавляем данный признак и к нашему кастомному чекбоксу  */
+.checkbox:checked + .checkbox-custom::before {
+    content: ""; /* Добавляем наш псевдоэлемент */
+    display: block; /* Делаем его блочным элементом */
```
он и так блочный по умолчанию

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,244 @@
+    padding: 14px 0 16px 0;
+    color: #939393;
+    font-size: 25px;
+    font-family: Verdana;
```
всегда нужно делать фолбек на стандартный шрифт sans или sans-serif, вдруг в системе верданы нет

ShevchenkoAlexen committed:
поправил код. подготовил к нескольким тодо

ShevchenkoAlexen committed:
добавлена поддержка создание нового тодолиста

ShevchenkoAlexen committed:
шрифты

ShevchenkoAlexen committed:
to classes

ShevchenkoAlexen committed:
to classes

ShevchenkoAlexen committed:
to classes

ShevchenkoAlexen committed:
Merge pull request #1 from ShevchenkoAlexen/по_другому

по другому

ShevchenkoAlexen committed:
добавлено удаление тодо листа

ShevchenkoAlexen committed:
на 2 файла

ShevchenkoAlexen committed:
добавил класс таск

ShevchenkoAlexen committed:
Merge pull request #2 from ShevchenkoAlexen/class_task

Class task

ShevchenkoAlexen committed:
exports

ShevchenkoAlexen committed:
exports

ShevchenkoAlexen committed:
Merge pull request #3 from ShevchenkoAlexen/class_task

exports

ShevchenkoAlexen committed:
Сделал сборку, допилил todolist  для возможности работы в мин функционале.

минимальный вид:
    <div name = "TodoList" class="body">
        <form class = "input_form">
            <input type="submit" name="sub_item" class="sub_button" value="&#9013;">
            <input type="text" class="input_section" placeholder="What need to be done?"/>

        </form>
            <div class="task template">
                    <span class="label"></span>
            </div>
    </div>

ShevchenkoAlexen committed:
Merge pull request #4 from ShevchenkoAlexen/class_task

Сделал сборку, допилил todolist

ShevchenkoAlexen committed:
new get set name

ShevchenkoAlexen committed:
Merge branch 'class_task'

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+    border-color: #efd4d4;
+}
+
+.unvisible, .template {
```
invisible

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+    float: right;
+}
+
+input[type="radio"] {
```
Всё же не очень хорошо завязываться на теги, а вдруг тебе на странице нужно будет сделать переключатели для чего-нибудь другого и всё сломается. Лучше завязывайся только на конкретные классы

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+}
+
+.body {
+    margin: 0 150px 50px 150px;
```
Делай выравнивание по центру экрана, а не отступы выставляй

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+.list_header {
+    box-shadow: 0 -4px 4px #f7f7f7 inset;
+    border-bottom: 1px solid #e6e6e6;
+    width: 1030px;
```
Тебе не нужно указывать ширину везде. div растягивается на 100% ширины контейнера, поэтому указывай ширину у базового блока, у остальных строй относительно её. Иначе, если ты захочешь сделать чуть поуже блок, то тебе придется менять везде ширину

EAndreyF replied:
Может получиться вот так, если всем задавать размеры: https://monosnap.com/file/UVXZXeBOrMCtZZo7jZK5OILd1EcnIq

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+    margin: 2px auto;
+    width: 950px;
+    font-weight: 100;
+    font-family: Verdana, sans;
```
Ниже тебе приходится много переопределять в этом классе, сделай его лучше более базовым. Лучше доопределять, чем переопределять

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+    font-size: 45px;
+    font-family: Verdana, sans;
+    padding: 0px 0px 0px 20px;
+    float: right;
```
Ты немного не на том уровне делаешь выравнивание.
Фактически у тебя блок `del_todo` отвечает за отображение крестика и возможно за его отступы. За позиционирование должен отвечать один из родителей.
Так ты сможешь переиспользовать этот элемент в других частях страницы.
И для других элементов так же

EAndreyF commented on index.html:
```diff
@@ -0,0 +1,73 @@
+
+        </div>
+        <form class = "input_form">
+            <input type="submit" name="sub_item" class="sub_button" value="&#9013;">
```
Когда кликаю на элемент появляется странное выделение
https://monosnap.com/file/wiOehjrQBpVz38drwiJW5yfptN7e1H

Вообще, ты не верно реализовал логику элемента. При клике на него должны выделяться все элементы
http://todomvc.com/examples/backbone/

Хотя, ты, наверное, не видел как должен работать функционал. Поэтому не знал, как должен работать. Но всёравно исправь, чтоыб работало как задумывалось:)

EAndreyF commented:
Текст не по центру кружка https://monosnap.com/file/ZGwSXUfZqiWH6Fci3dgg5070sUwRlq

EAndreyF commented on index.css:
```diff
@@ -0,0 +1,293 @@
+}
+
+.checkbox-custom {
+    margin: 0 0 0 20px;
```
Его можно сделать псевдоэлементов ::before, ::after у чекбокса, тогда верстка будет более лаконична, не нужен лишний дом элемент

EAndreyF commented on js.js:
```diff
@@ -0,0 +1,20 @@
+window.onload = function () {
+    // проверяем поддержку
+    var startTodo = [];
+    var lists = document.getElementsByName('TodoList');
```
Это странный селектор. Используй стандартный селектор по классу.

EAndreyF commented on js.js:
```diff
@@ -0,0 +1,20 @@
+    var startTodo = [];
+    var lists = document.getElementsByName('TodoList');
+    for (var i = 0; i < lists.length; i++) {
+        startTodo[i] = new TodoList(lists[i]);
```
Непонятно, зачем ты сохраняешь в массив, если ты потом с ним ничего не делаешь, можешь сразу добавить тестовую таску.

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+'use strict';
+
+class Task {
```
У тебя это отдельный класс, вынеси его в отдельный файл

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+
+}
+
+module.exports = class TodoList {
```
чуть более красиво, если сделать
```
class TodoList {
}

module.exports = TodoList;
```

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+     */
+    constructor(bodyElement) {
+        this.list = bodyElement;
+        this.changeViewButtonsElem = bodyElement.querySelectorAll('.radiokbox');
```
Старайся использовать правильные орфографически имена `radiokbox` -> `radiobox`

EAndreyF replied:
Название ещё плохо тем, что визуально я не могу его найти. Здорово придумывать такие имена, когда ты не заглядывая в верстку можешь найти, например `viewtype-button`. Ты же переменные нормально называешь, так же называй и CSS класы

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+     * @param {String}view
+     * @returns {TodoList}
+     */
+    show(view) {
```
Можно сделать немного симпатичнее:
```js
if (view === 'complited') {
 return this.showComplited();
} else if (view === 'active') {
 return this.showActive();
}

return this.showAll();
```

Тут ты сразу понимаешь что у тебя заканчивается функция внутри if и тебе не нужно читать код ниже.

ShevchenkoAlexen replied:
у меня 3 варианта,  а условия - 2 .  Если добавить еще елсе, то будет работать

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+    addListnerRadio() {
+        for (var i = 0; i < this.changeViewButtonsElem.length; i++) {
+            var t = this;
+            this.changeViewButtonsElem[i].addEventListener('click', function () {
```
Используй стрелочную функцию, а текущее значение получай через `e.target.value`

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+     */
+    addListnerInputForm() {
+        var t = this;
+        var nt = this.inputForm.querySelector('.input_section');
```
плохие имена переменных) придумай лучше.
Для сохранения `this`, давай использовать переменную `_this`
Но, если использовать стрелочные функции, то this не придется сохранять

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+
+    /**
+     * Добавдение новой таски в тодолист
+     * @param {String}newT
```
давай везде будем добавлять отступ после закрывающей }

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+        }
+        var t = new Task(this, newT, false);
+        this.tasks.push(t);
+        this.template.parentNode.appendChild(t.task); // Вставляем объект
```
Ты же вставляешь html, давай переменную назовем приближенно к этому, а то получается как `task.task`.
Да и название переменной `t` - плохое

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+    changeCount() {
+        var count = this.tasks.length;
+        for (var i = count - 1; i >= 0; i--) {
+            if (!this.tasks[i].task.parentNode) {
```
А можно как-то явно удалять, после явного удаления, а не через постфактум проверки, что из DOM исчезло, значит можно из массива удалить

EAndreyF replied:
И вообще, ты тут лезешь во внутреннюю структуру таски, не очень хорошо. Если хочешь проверить на видимость, вызови функцию `isVisible`

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+        }
+        count = this.tasks.length;
+        var elemCount = this.list.querySelector('.count');
+        if (!elemCount) {
```
А такое может быть?

ShevchenkoAlexen replied:
может, если счетчик не используется.
наверно правильнее будет сделать отдельный класс для счетчика, в который передавать блок в котором искать, и класс по которому считать

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+                this.tasks.splice(i, 1);
+            }
+        }
+        count = this.tasks.length;
```
одной и той же переменной присваивается одно и тоже значение (визуально) - не очень хорошо. Понятно, что ты выше изменил длинну массива, но давай придумаем лучше способ

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+    addListnerButtonClear() {
+        var t = this;
+        var clButton = this.list.querySelector('.button_clear');
+        if (!clButton) {
```
А такое может быть? Не нужно добавлять лишних проверок если всё управляется данным классом

ShevchenkoAlexen replied:
Эти проверки сделаны чтобы можно было удалить этот элемент из тудушки, и все по прежнему работало.

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+     */
+    showComplited() {
+        for (var i = 0; i < this.tasks.length; i++) {
+            var checked = this.tasks[i].check;
```
Можно проще `this.tasks[i].viewStatus = this.tasks[i].check`.
Но тогда это получается странно, что мы в родительском классе полностью оперируем свойствами из внутреннего класса. Но может и норм, но странно:)

ShevchenkoAlexen committed:
поправил "текст по середине кружка"

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+     */
+    createNewTodo(type) {
+        var count = document.getElementsByName('TodoList').length;
+        var div = this.list.cloneNode(true);
```
Попробуй написать собственный шаблонизатор или придумай механизм создания todo, не путем копирования текущего, а создания из шаблона нового

EAndreyF replied:
И функция получится гораздо проще

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+        this.addListnerDelLine();
+    }
+
+    get taskNode() {
```
круто придумал с геттерами. Но не забывай соблюдать баланс, чтобы не терять общее понимание

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+
+    set taskNode(task) {
+        this.task = task;
+        this.check = task.querySelector('.checkbox').checked;
```
`this.isChecked` - лучше иллюстрирует предназначение

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+
+    set viewStatus(stat) {
+        if (!stat) {
+            this.task.style.display = 'none';
```
Всегда лучше делать через классы, чем через прямые стили. Так что меняй класс, а не стиль

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+            return this;
+        }
+        this.checkBox.addEventListener('click', function () {
+            var l = this.closest('.todo_item').querySelector('.label');
```
Плохая переменная

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+    addListnerDelLine() {
+        var t = this;
+        var dItem = this.task.querySelectorAll('.del_item_button');
+        for (var i = 0; i < dItem.length; i++) {
```
У тебя же одна такая кнопка, так что цикл не нужен

ShevchenkoAlexen replied:
а вдруг будет 2 кнопки. вначале и в конце строки)

EAndreyF commented on todolist.js:
```diff
@@ -0,0 +1,406 @@
+        for (var i = 0; i < dItem.length; i++) {
+            dItem[i].addEventListener('click', function () {
+                t.deleteTask();
+                t.todo.changeCount();
```
Давай сделаем немного по другому. Приближенно к тому, что придется делать, когда будем писать на реакте.

Пускай в конструктор передаются функции, которые мы можем вызвать и тут мы только эти функции вызываем, а не обращаемся напрямую к инстансу родителя. Подробнее можешь прочитать https://medium.com/@thejasonfile/callback-functions-in-react-e822ebede766

EAndreyF replied:
Тогда тебе не придется лазить в разметку родителя через `parentNode`

ShevchenkoAlexen committed:
webpack

ShevchenkoAlexen committed:
Merge pull request #5 from ShevchenkoAlexen/bem

webpack

ShevchenkoAlexen committed:
webpack

ShevchenkoAlexen committed:
Merge remote-tracking branch 'origin/master'

ShevchenkoAlexen committed:
webpack

ShevchenkoAlexen committed:
Merge pull request #6 from ShevchenkoAlexen/bem

webpack

ShevchenkoAlexen committed:
webpack

ShevchenkoAlexen committed:
webpack

ShevchenkoAlexen committed:
Delete build.js

ShevchenkoAlexen committed:
Merge pull request #7 from ShevchenkoAlexen/bem

надеюсь взлетит
